backend/routes/chatRouter.js

```diff
@@ -1,33 +1,80 @@
 const express = require("express");
 const path = require("path");
 const chatRouter = express.Router();
+const Chat = require("../models/Chat.js");
+
+// Join a chat
+chatRouter.get("/join/:chat_id", async (req, res) => {
+	console.log("GET request recieved");
+    const chatId = req.params.chat_id;
+
+    try {
+		//Check if a chat with the given PIN exists
+		console.log("Attempting to find chat");
+        const chat = await Chat.findOne({ pin: chatId });
+		console.log("Chat found");
+
+        if (chat) {
+			//Respond true if the chatroom exists
+			res.json({ exists: true });
```

Comment:
I think here we also need to add the joined user into the chat  Users field

Reply:
Since that this point, we still don't have a way to determine who that user is (guest or not), i will let the `chat/:chat_id` handles this (it will determine if that user is guest and create new User document accordingly and associate the chat to it)

Reply:
For now i just handle this for guest user, but when we use cookie we can easily determine that user's identify